flaskdeid/deid.py

```diff
@@ -13,10 +13,33 @@
 bp = Blueprint('deid', __name__, url_prefix='/deid')
 
 @bp.route("/annotate", methods=['POST'])
-def identify_phi(**kwargs):
+def annotate(**kwargs):
+    if not request.json or not 'extract_text' in request.json:
+        abort(400)
+
+    note_text = request.json['extract_text']
+    if note_text:
+        return identify_phi(note_text, **kwargs)
+    else:
+        msg = "No Entity Text was found"
+        logger.info("No entities returned")
+        return Response(msg, status=400)
+
+def identify_phi(note_text, **kwargs):
     annotations = []
-    annotations += [AnnotationFactory.from_medlp(phi) for phi in medlp.annotate_phi(**kwargs)]
-    annotations += [AnnotationFactory.from_hutchner(phi) for phi in hutchner.annotate_phi(**kwargs)]
+    try:
+        annotations += [AnnotationFactory.from_medlp(phi) for phi in medlpInterface.get_phi(note_text)]
+    except ValueError as e:
+        msg = "An error occurred while calling MedLP"
+        logger.warning("An error occurred while calling MedLPInterface: {}".format(e))
+        return Response(msg, status=400)
+
+    try:
+        annotations += [AnnotationFactory.from_hutchner(phi) for phi in hutchNERInterface.predict(note_text, **kwargs).NER_token_labels]
+    except ValueError as e:
+        msg = "An error occurred while calling HutchNER"
+        logger.warning("An error occurred while calling HutchNERInterface: {}".format(e))
```

Comment:
What about just `logger.warning("{}: {}".format(msg, e))`? Just to cut down slightly on duplicating strings/code.